File: atcoder/abc180B_various_distances.nim
# unihernandez22
# https://atcoder.jp/contests/abc180/tasks/abc180_b
# implementation

import strutils, sequtils, math

discard stdin.readLine
let x = stdin.readLine.split.map parseInt

echo x.map(proc (i: int): int = i.abs).sum
echo x.map(proc (i: int): int = i.abs ^ 2).sum.float.sqrt
echo x.map(proc (i: int): int = i.abs).max

File: atcoder/abc180B_various_distances.c
// Vicfred
// https://atcoder.jp/contests/abc180/tasks/abc180_b
// math
#include <stdio.h>
#include <math.h>

int main() {
    int n;
    scanf("%d\n", &n);

    long long int manhattan = 0L;
    double euclidean = 0.0L;
    long long int chebyshev = 0L;

    long long int x;
    for(int i = 0; i < n; i++) {
        scanf("%lld", &x);
        x = fabs(x);
        manhattan += x;
        euclidean += x*x;
        chebyshev = fmax(chebyshev, x);
    }

    euclidean = sqrt(euclidean);

    printf("%lld\n%.12lf\n%lld\n", manhattan, euclidean, chebyshev);
    
    return 0;
}
